examples/uniswap-from-near/contract/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+use aurora_sdk::{
+    ethabi, near_sdk, Address, CallArgs, FunctionCallArgsV1, SubmitResult, TransactionStatus, U256,
+};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::U64;
+use near_sdk::{env, near_bindgen, AccountId, PanicOnDefault, Promise};
+
+const DEFAULT_FEE: u64 = 500;
+/// Selector for [exactOutputSingle](https://docs.uniswap.org/contracts/v3/reference/periphery/SwapRouter#exactoutputsingle)
+const EXACT_OUTPUT_SINGLE_SELECTOR: [u8; 4] = [219, 62, 33, 152];
+
+#[near_bindgen]
+#[derive(PanicOnDefault, BorshDeserialize, BorshSerialize)]
+pub struct UniswapProxy {
+    aurora: AccountId,
+    uniswap: Address,
+}
+
+#[near_bindgen]
+impl UniswapProxy {
+    /// Function to initialize this contract with the account ID of the Aurora EVM and the
+    /// address for the Uniswap router contract deployed there.
+    #[init]
+    pub fn new(aurora: AccountId, uniswap_address: String) -> Self {
+        Self {
+            aurora,
+            uniswap: aurora_sdk::parse_address(&uniswap_address).unwrap(),
+        }
+    }
+
+    /// Proxy for [exactOutputSingle](https://docs.uniswap.org/contracts/v3/reference/periphery/SwapRouter#exactoutputsingle).
+    /// Calls the `exact_output_single` function of a Uniswap router contract deployed on Aurora.
+    pub fn exact_output_single(&self, params: SerializableExactOutputSingleParams) -> Promise {
+        let params: ExactOutputSingleParams = params.try_into().unwrap();
```

Comment:
I definitely agree something like this would be nice! I've captured it [as an issue](https://github.com/aurora-is-near/aurora-contracts-sdk/issues/3) for now and we can tackle it in the future when we have more time.